src/scales/scale.time.js

```diff
@@ -36,10 +40,94 @@ module.exports = function(Chart) {
 			},
 		},
 		ticks: {
-			autoSkip: false
+			autoSkip: false,
+			mode: 'series',   // 'linear|series'
```

Comment:
Seems you right, `auto + series` and `auto + linear` don't do the same if leading / trailing times are adjusted. Still think `auto + series` would be better defaults but it's a breaking change, so should be: `auto + linear`.